library/std/src/prelude/v1.rs

```diff
@@ -43,15 +43,44 @@ pub use crate::option::Option::{self, None, Some};
 #[doc(no_inline)]
 pub use crate::result::Result::{self, Err, Ok};
 
-// Re-exported built-in macros
+// Re-exported built-in macros and traits
 #[stable(feature = "builtin_macro_prelude", since = "1.38.0")]
 #[doc(no_inline)]
+#[allow(deprecated)]
 pub use core::prelude::v1::{
-    assert, cfg, column, compile_error, concat, env, file, format_args,
-    format_args_nl, include, include_bytes, include_str, line, log_syntax, module_path, option_env,
-    stringify, trace_macros, Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd,
+    assert, assert_eq, assert_ne, cfg, column, compile_error, concat, debug_assert, debug_assert_eq, debug_assert_ne, env, file, format_args, include, include_bytes, include_str, line, matches,
+    module_path, option_env, stringify, todo, r#try, unimplemented, unreachable, write,
+    writeln, Clone, Copy, Debug, Default, Eq, Hash, Ord, PartialEq, PartialOrd,
 };
 
+#[stable(feature = "builtin_macro_prelude", since = "1.38.0")]
+#[doc(no_inline)]
+pub use crate::{
+    dbg, eprint, eprintln, format, is_x86_feature_detected, print, println, thread_local
+};
+
+// These macros needs special handling, so that we don't export it *and* the modules of the same
+// name. We only want the macro in the prelude.
+mod ambiguous_macro_only_std {
+    #[allow(hidden_glob_reexports)]
+    mod vec {}
+    #[allow(hidden_glob_reexports)]
+    mod panic {}
+    // Building std without the allow exported_private_dependencies will create warnings, but then
+    // clippy claims its a useless_attribute. So silence both.
+    #[allow(clippy::useless_attribute)]
+    #[allow(exported_private_dependencies)]
```

Comment:
```suggestion
    #[expect(exported_private_dependencies)]
```
This may indicate an issue in `exported_private_dependencies`, better to remove it if the issue is resolved.

In fact, all `allow`s added in this PR can be replaced with `expect`s.